### PR TITLE
Fix duplicate symbol linkage error

### LINKED ISSUE #62831: build failed : ld.lld: error: duplicate symbol: DB::registerFunctionHas
**ClickHouse version:**
v24.3.2.23-lts

**Operating system**
> CentOS7 Linux  3.10.0-1160.11.1.el7.x86_64 x86_64 GNU/Linux

**Cmake version**
cmake version 3.20.6
**Ninja version**
ninja 1.10.0
**Compiler name and version**
clang-18
**Full cmake and/or ninja output**
```
ld.lld: error: duplicate symbol: DB::registerFunctionHas(DB::FunctionFactory&)
>>> defined at has.cpp:11 (./build/../src/Functions/array/has.cpp:11)
>>>            has.cpp.o:(DB::registerFunctionHas(DB::FunctionFactory&)) in archive src/libdbms.a
>>> defined at has.cpp
>>>            src/Functions/array/CMakeFiles/clickhouse_functions_array.dir/./has.cpp.o:(.text+0x0)

ld.lld: error: duplicate symbol: DB::createInternalFunctionHasOverloadResolver()
>>> defined at has.cpp:14 (./build/../src/Functions/array/has.cpp:14)
>>>            has.cpp.o:(DB::createInternalFunctionHasOverloadResolver()) in archive src/libdbms.a
>>> defined at has.cpp
>>>            src/Functions/array/CMakeFiles/clickhouse_functions_array.dir/./has.cpp.o:(.text+0x80)
clang-18: error: linker command failed with exit code 1 (use -v to see invocation)
```

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [x]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

---

```
ld.lld-17: error: duplicate symbol: DB::registerFunctionHas(DB::FunctionFactory&)
>>> defined at has.cpp
>>>            has.cpp.o:(DB::registerFunctionHas(DB::FunctionFactory&)) in archive src/libdbms.a
>>> defined at has.cpp
>>>            src/Functions/array/CMakeFiles/clickhouse_functions_array.dir/./has.cpp.o:(.text+0x0)

ld.lld-17: error: duplicate symbol: DB::createInternalFunctionHasOverloadResolver()
>>> defined at has.cpp
>>>            has.cpp.o:(DB::createInternalFunctionHasOverloadResolver()) in archive src/libdbms.a
>>> defined at has.cpp
>>>            src/Functions/array/CMakeFiles/clickhouse_functions_array.dir/./has.cpp.o:(.text+0x80)
```

It happens when I build `clickhouse` and `unit_tests_dbms` targets ATST.

Fixes: #62831

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 89a8925f80c1174ce8546758cc8155cb53805b00 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65705/89a8925f80c1174ce8546758cc8155cb53805b00/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Cloud fork sync (only for ClickHouse Inc. employees)</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by Algunenano** [comment]
Related https://github.com/ClickHouse/ClickHouse/issues/62831 ?

**Comment by nickitat** [comment]
> Related #62831 ?

yes, looks similar

File: src/Functions/CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,6 @@ set(DBMS_FUNCTIONS
     multiIf.cpp
     multiMatchAny.cpp
     checkHyperscanRegexp.cpp
-    array/has.cpp
```
Comment (Algunenano) [inline src/Functions/CMakeLists.txt:16]: At this point, if we are linking `clickhouse_functions` everywhere, do we even need these blocks that moves files or headers to `dbms`? Maybe it'd be simpler to just use clickhouse_functions eveywhere

Comment (nickitat) [reply src/Functions/CMakeLists.txt:16]: i spent quite some time refusing to check for this, but yes

https://github.com/ClickHouse/ClickHouse/blob/8b20e9f505f5faaac3690c24c2f4f69dd6bbbe80/src/Functions/tests/gtest_ternary_logic.cpp#L11-L12

Comment (Algunenano) [reply src/Functions/CMakeLists.txt:16]: Solution: Remove this file

Comment (nickitat) [reply src/Functions/CMakeLists.txt:16]: i made an attempt to fix it, but there were too many dependencies from the source file that i didn't want to move to the header

**Review by Algunenano — APPROVED** [review]

**Comment by jrdi** [comment]
macOS ARM "native" build started failing after this PR. I'm getting the following error on `clickhouse-library-bridge` executable linking step

```
duplicate symbol '_explicit_bzero' in:
    contrib/libpq-cmake/lib_libpq.a[49](explicit_bzero.c.o)
    contrib/libssh-cmake/lib_ssh.a[33](misc.c.o)
ld: 1 duplicate symbols
```

I'm able to build it by disabling either libpq or ssh. Is there anything I can do to help fix this?

**Pull request merged**